extensions-core/multi-stage-query/src/test/java/org/apache/druid/msq/test/CalciteSelectQueryTestMSQ.java

```diff
@@ -0,0 +1,369 @@
+              .withDimensionsSpec(
+                  new DimensionsSpec(
+                      ImmutableList.of(
+                          new StringDimensionSchema("dim1"),
+                          new StringDimensionSchema("dim2"),
+                          new LongDimensionSchema("dim3")
+                      )
+                  )
+              )
+              .withMetrics(
+                  new CountAggregatorFactory("cnt"),
+                  new LongSumAggregatorFactory("m1", "m1"),
+                  new DoubleSumAggregatorFactory("m2", "m2"),
+                  new HyperUniquesAggregatorFactory("unique_dim1", "dim1")
+              )
+              .withRollup(false)
+              .build();
+          index = IndexBuilder
+              .create()
+              .tmpDir(new File(temporaryFolder.newFolder(), "1"))
+              .segmentWriteOutMediumFactory(OffHeapMemorySegmentWriteOutMediumFactory.instance())
+              .schema(indexSchemaDifferentDim3M1Types)
+              .rows(ROWS2)
+              .buildMMappedIndex();
+          break;
+        default:
+          throw new ISE("Cannot query segment %s in test runner", segmentId);
+
+      }
+    }
+    catch (IOException e) {
+      throw new ISE(e, "Unable to load index for segment %s", segmentId);
+    }
+    Segment segment = new Segment()
+    {
+      @Override
+      public SegmentId getId()
+      {
+        return segmentId;
+      }
+
+      @Override
+      public Interval getDataInterval()
+      {
+        return segmentId.getInterval();
+      }
+
+      @Nullable
+      @Override
+      public QueryableIndex asQueryableIndex()
+      {
+        return index;
+      }
+
+      @Override
+      public StorageAdapter asStorageAdapter()
+      {
+        return new QueryableIndexStorageAdapter(index);
+      }
+
+      @Override
+      public void close()
+      {
+      }
+    };
+    return new Supplier<Pair<Segment, Closeable>>()
+    {
+      @Override
+      public Pair<Segment, Closeable> get()
+      {
+        return new Pair<>(segment, Closer.create());
+      }
+    };
+  }
+
+  protected Map<String, Object> defaultScanQueryContext(final RowSignature signature)
+  {
+    try {
+      return ImmutableMap.<String, Object>builder()
+                         .putAll(MSQTestBase.DEFAULT_MSQ_CONTEXT)
+                         .put(
+                             DruidQuery.CTX_SCAN_SIGNATURE,
+                             queryFramework().queryJsonMapper().writeValueAsString(signature)
+                         )
+                         .build();
+    }
+    catch (JsonProcessingException e) {
+      throw new RuntimeException(e);
+    }
+  }
+
+
+
+  @Test
+  public void testOrderThenLimitThenFilter()
+  {
+    RowSignature resultSignature = RowSignature.builder()
+                                               .add("dim1", ColumnType.STRING)
+                                               .build();
+
+    testQueryWithMSQ(
```

Comment:
This test uses an MSQ-specific test function. The original goal was to reuse the existing tests. Is this an intermediate step along the way? Or, have we opted to copy/paste tests to the MSQ form?

Reply:
Added a top level comment, and pushing out the code for the same